XmlFormatter/src/Formatter/XmlFormatterProvider.cs

```diff
@@ -0,0 +1,125 @@
+        /// </summary>
+        private readonly string name;
+
+        /// <inheritdoc/>
+        public string Name => name;
+
+        /// <summary>
+        /// The extension supported by this formatter
+        /// </summary>
+        private readonly string extension;
+
+        /// <inheritdoc/>
+        public string Extension => extension;
+
+        /// <summary>
+        /// Create a new instance of this class
+        /// </summary>
+        public XmlFormatterProvider()
+        {
+            name = "XmlFormatter";
+            extension = "xml";
+        }
+
+        /// <inheritdoc/>
+        public bool ConvertToFormat(string filePath, string outputName, ModesEnum mode)
+        {
+            switch (mode)
+            {
+                case ModesEnum.Formatted:
+                    return ConvertToFormatted(filePath, outputName);
+                case ModesEnum.Flat:
+                    return ConvertToFlat(filePath, outputName);
+                default:
+                    break;
+            }
+            return false;
+        }
+
+        /// <inheritdoc/>
+        public bool ConvertToFlat(string filePath, string outputName)
+        {
+            FormatFile(filePath, outputName, SaveOptions.DisableFormatting);
+            return true;
+        }
+
+        /// <inheritdoc/>
+        public bool ConvertToFormatted(string filePath, string outputName)
+        {
+            FormatFile(filePath, outputName, SaveOptions.None);
+            return true;
+        }
+
+        /// <summary>
+        /// This method will convert the file 
+        /// </summary>
+        /// <param name="inputFilePath">The input file to convert</param>
+        /// <param name="outputName">The output file to generate</param>
+        /// <param name="options">The save options to use</param>
+        private async void FormatFile(string inputFilePath, string outputName, SaveOptions options)
+        {
+            try
+            {
+                using (FileStream fileStream = File.Open(inputFilePath, FileMode.Open, FileAccess.ReadWrite, FileShare.None))
+                {
+                    fileStream.Close();
+                }
+                if (File.Exists(outputName))
+                {
+                    using (FileStream fileStream = File.Open(outputName, FileMode.Open, FileAccess.ReadWrite, FileShare.None))
+                    {
+                        fileStream.Close();
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                FireEvent("Saving failed", "Files where locked!");
+                return;
+            }
+
+            FireEvent("Loading", "Loading ...");
+            XElement fileToConvert = await Task<XElement>.Run(() =>
+            {
+                return XElement.Load(inputFilePath);
+            });
+
+            FireEvent("Saving", "Saving ...");
+            await Task.Run(() => fileToConvert.Save(outputName, options));           
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5053278)